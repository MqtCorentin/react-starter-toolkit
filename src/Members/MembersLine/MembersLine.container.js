import React from 'react';
import MembersLine from './MembersLine';

export const MembersLineEnhanced = props => <MembersLine {...props} />;

export default MembersLineEnhanced;
