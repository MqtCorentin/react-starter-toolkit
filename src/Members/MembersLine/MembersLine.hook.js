import { useState, useCallback } from 'react';

export const useToggleModalDuplicate = (initState = false) => {
  const [isOpenModalDuplicate, setStateToggleModalDuplicate] = useState(initState);

  const closeModalDuplicate = useCallback(() => {
    setStateToggleModalDuplicate(false);
  }, []);

  const openModalDuplicate = useCallback(() => {
    setStateToggleModalDuplicate(true);
  }, []);

  return { closeModalDuplicate, openModalDuplicate, isOpenModalDuplicate };
};

export const useToggleModalDelete = (initState = false) => {
  const [isOpenModalDelete, setStateToggleModalDelete] = useState(initState);

  const closeModalDelete = useCallback(() => {
    setStateToggleModalDelete(false);
  }, []);

  const openModalDelete = useCallback(() => {
    setStateToggleModalDelete(true);
  }, []);

  return { closeModalDelete, openModalDelete, isOpenModalDelete };
};
