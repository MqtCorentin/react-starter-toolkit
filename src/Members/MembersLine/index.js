export { default } from './MembersLine.container';
