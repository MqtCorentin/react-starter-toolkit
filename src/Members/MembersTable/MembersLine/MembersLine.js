import React from 'react';
import PropTypes from 'prop-types';
import { Table } from '@axa-fr/react-toolkit-all';

const MembersLine = ({ firstname, lastname, birthdate, sexe }) => (
  <Table.Tr>
    <Table.Td>{firstname}</Table.Td>
    <Table.Td>{lastname}</Table.Td>
    <Table.Td>{birthdate}</Table.Td>
    <Table.Td>{sexe}</Table.Td>
  </Table.Tr>
);

MembersLine.propTypes = {
  firstname: PropTypes.string.isRequired,
  lastname: PropTypes.string.isRequired,
  birthdate: PropTypes.string.isRequired,
  sexe: PropTypes.string.isRequired,
};

export default MembersLine;
