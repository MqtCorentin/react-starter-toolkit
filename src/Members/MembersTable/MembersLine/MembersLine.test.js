import React from 'react';
import { render } from '@testing-library/react';
import MembersLine from './MembersLine';

const defaultProps = {
  _id: '00001',
  type: 'PP',
  firstname: 'Samuel',
  lastname: 'Gomez',
  birthdate: '1985-10-20T13:44:20.540000',
  created: '2020-10-20T13:44:20.540000',
  fetch: jest.fn(),
};

describe('<MembersLine/>', () => {
  it('Should render MembersLine', () => {
    const tbody = document.createElement('tbody');
    const { asFragment } = render(<MembersLine {...defaultProps} />, {
      container: document.body.appendChild(tbody),
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
