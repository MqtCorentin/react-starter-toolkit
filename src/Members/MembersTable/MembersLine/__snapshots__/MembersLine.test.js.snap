// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MembersLine/> Should render MembersLine 1`] = `
<DocumentFragment>
  <tr
    class="af-table__tr"
  >
    <td
      class="af-table__cell"
    >
      Samuel
    </td>
    <td
      class="af-table__cell"
    >
      Gomez
    </td>
    <td
      class="af-table__cell"
    >
      1985-10-20T13:44:20.540000
    </td>
    <td
      class="af-table__cell"
    />
  </tr>
</DocumentFragment>
`;
